### PR TITLE
Support new 1.16 metrics with filter to exclude deprecated metrics

### PR DESCRIPTION
**Pull request by maimaisie**
###### Description

This PR brings back changes in #372 and adds a grep filter plugin to exclude the deprecated metrics by checking the k8s version (which is a built-in object in helm)

###### Testing performed

- [ ] ci/build.sh
- [ ] Redeploy fluentd and fluentd-events pods
- [x] Confirm events, logs, and metrics are coming in

### CODE REVIEW

File: deploy/helm/sumologic/conf/metrics/metrics.conf [hunk 1]
```diff
@@ -10,6 +10,15 @@
   @label @DATAPOINT
 </match>
 <label @DATAPOINT>
+{{- if and (eq .Capabilities.KubeVersion.Major "1") (eq .Capabilities.KubeVersion.Minor "16") }}
```
Comment (rvmiller89) [inline deploy/helm/sumologic/conf/metrics/metrics.conf]: what about 1.17 and above? Or these metrics only apply for 1.16?

Comment (maimaisie) [reply deploy/helm/sumologic/conf/metrics/metrics.conf]: I think they are getting removed in 1.17
> Beta release target 1.17
All previously marked deprecated metrics will be removed from the codebase.

in https://github.com/kubernetes/enhancements/issues/1206

Comment (frankreno) [reply deploy/helm/sumologic/conf/metrics/metrics.conf]: They are getting removed, but i believe both metrics co-exist in 1.14, 1.15 and 1.16

Comment (frankreno) [reply deploy/helm/sumologic/conf/metrics/metrics.conf]: https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG-1.14.md#deprecated-metrics -  you can see that they deprecated the old and added the new in 1.14.  So, i think this version check should be for 1.14 and above.

Comment (maimaisie) [reply deploy/helm/sumologic/conf/metrics/metrics.conf]: I just realized that too, let me update the condition. Thanks!

Comment (maimaisie) [reply deploy/helm/sumologic/conf/metrics/metrics.conf]: hmm, what should we do for the non-helm yaml file? it looks like CI added the filter to the generated file by helm template

**Comment by samjsong** [comment]
@maimaisie I just noticed a binary file tgz file got added to the PR, that's not expected is it?

**Comment by maimaisie** [comment]
oh oops, thanks for noticing that! removed

**Pull request merged**